examples/the-ultimate-guide-to-pull-requests-and-code-reviews/README.md

```diff
@@ -0,0 +1,252 @@
+
+## Additional Tips for Code Reviews
+
+- **Limit Review Duration:** Keep sessions to 1 hour to avoid fatigue.
+- **Focus on Small Code Blocks:** Review 200-400 lines at a time.
+- **Actionable Feedback:** Provide examples to illustrate suggestions.
+- **Verify Bug Fixes:** Test changes in a live environment if possible.
+- **Use Templates:** Ensure all necessary information is consistently included.
+
+## Pull Request Templates
+
+Most Git hosting platforms, such as GitHub, support pull request templates. Pull request templates help standardize the information provided in a PR, making it easier for reviewers to understand the changes and context.
+
+Using pull request templates ensures that all necessary information is provided consistently.
+
+For example, on GitHub, you can create a `PULL_REQUEST_TEMPLATE.md` file in the `.github` directory of your repository. This template can include sections for description, type of change, testing instructions, checklist, etc.
+
+**Example Pull Request Template:**
+
+```markdown
+## Description
+
+Please include a summary of the changes and the related issue. List any dependencies that are required for this change.
+
+Fixes # (issue)
+
+## Type of Change
+
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+- [ ] Documentation update
+
+## How Has This Been Tested?
+
+Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.
+
+- [ ] Test A
+- [ ] Test B
+
+## Checklist:
+
+- [ ] My code follows the style guidelines of this project
+- [ ] I have performed a self-review of my code
+- [ ] I have commented my code, particularly in hard-to-understand areas
+- [ ] I have made corresponding changes to the documentation
+- [ ] My changes generate no new warnings
+- [ ] Any dependent changes have been merged and published in downstream modules
+- [ ] I have checked my code and corrected any misspellings
+```
+
+This template ensures that all PRs include essential information and follow the same structure.
+
+In the section below, we provide a detailed guide on how to [automate PR validation using DangerJS and TypeScript](#example-automate-pr-validation-with-dangerjs-and-typescript).
+
+You could also use templates to manage different aspects of the Issue and PR lifecycle. For example, you could have a template for bug reports, feature requests, and other types of issues. You can check out one complete example of how to manage this in the [Issues section of the DevOps Reference repository](https://github.com/nanlabs/devops-reference/issues/new/choose) where we have templates for different types of issues. You can also check out we made it possible in the [source of the issue templates](https://github.com/nanlabs/devops-reference/tree/main/.github/ISSUE_TEMPLATE).
```

Comment:
When I enter to any template I see a gray forbidden box. Is that ok?

![image](https://github.com/nanlabs/devops-reference/assets/5368300/344b27be-7909-460f-8ffa-6cceac202fe7)